torch/nn/utils/_expanded_weights/expanded_weights_utils.py

```diff
@@ -90,6 +91,15 @@ def set_grad_sample_if_exists(maybe_expanded_weight, per_sample_grad_fn):
     unpacked = unpack_expanded_weight_or_tensor(maybe_expanded_weight)
     if isinstance(maybe_expanded_weight, ExpandedWeight):
         grad_sample_contribution = maybe_scale_by_batch_size(per_sample_grad_fn(unpacked), maybe_expanded_weight)
+
+        if maybe_expanded_weight.batch_size > grad_sample_contribution.shape[0]:
+            # this only passes the other checks if the arg allows smaller batch sizes
+            int = torch.zeros(maybe_expanded_weight.batch_size, *grad_sample_contribution.shape[1:],
+                              dtype=grad_sample_contribution.dtype,
+                              device=grad_sample_contribution.device)
+            int[:grad_sample_contribution.shape[0]] = grad_sample_contribution
+            grad_sample_contribution = int
```

Comment:
nit: `int` is special in Python, let's rename it and not shadow

Reply:
Done!